aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -458,11 +459,46 @@ Tensor & masked_scatter__cuda(Tensor& self, const Tensor& mask, const Tensor& so
   return self;
 }
 
+template <typename scalar_t>
+void flip_kernel_impl(TensorIterator& iter) {
+  if (!iter.can_use_32bit_indexing()) {
+    for (auto& sub_iter : iter.with_32bit_indexing()) {
```

Comment:
`with_32_bit_indexing()` is using `get_dim_to_split` function https://github.com/pytorch/pytorch/blob/9fc0c5a54a17689dd92029347473df69ece1f409/aten/src/ATen/TensorIterator.cpp#L1087 that, as you can see, won't work correctly with negative strides. 
So there are 2 options here: think about how it can be modified, or instantiate 64-bit kernel and offset calculator for large tensors. Unfortunately, we don't routinely run tests for large tensors, so this was not flagged.